src/Updater.js

```diff
@@ -1,154 +1,117 @@
-            this.saga,
-            actionPrefix,
-            dispatch
-          );
-        });
-      } else if (action && action.type === Unmount &&
-          action.effectExecutor && sagaRepository[this.getActionPrefix(action)]) {
-        const actionPrefix = this.getActionPrefix(action);
-
-        if (!sagaRepository[actionPrefix].isCancelled()) {
-          sagaRepository[actionPrefix].cancel();
-        }
-        delete sagaRepository[actionPrefix];
-      }
+    const {
+      saga,
+      SagaAbstraction
+    } = this;
 
+    return (model, action) => {
       if (action) {
-        // Matching logic is fairly simple
-        // it just maps over all the provided matchers and tries matching the action
-        // then only trutrhy matches pass
-        const matchedMatchers = this.matchers
-          .map(({ matcher, updater }) => ({ match: matcher(action), updater }))
-          .filter(({ match }) => !!match);
-
-          // Calling the appropriate updater
-          // Effect executor is passed to the Updater so that it can be used
-          // for composition
-        const reduction = matchedMatchers
-          .reduce((partialReduction, { match: { wrap, args, unwrap }, updater }) => updater(
-            partialReduction,
-            { ...action, type: unwrap, args, wrap: `${(action.wrap || '')}${wrap}` }
-          ), model);
-
-        // If there is an existing Saga instance for the updater
-        // Store reduction into State Repository and notify
-        // all subscribers for the specific Saga instance
-        if (this.saga && action.effectExecutor) {
-          const actionPrefix = this.getActionPrefix(action);
-
-          action.effectExecutor(() => {
-            if (subscribersRepository[actionPrefix]) {
-              stateRepository[actionPrefix] = reduction;
-              subscribersRepository[actionPrefix].forEach(({ cb }) => cb(action));
+        const {
+          sagaRepository,
+          effectExecutor
+        } = action;
+
+        const sagaId = action.matching ? action.matching.id : '';
+        const mutatedModel = reducer(model, action);
+
+        if (effectExecutor) {
+          effectExecutor(dispatch => {
+            if (action.type === Mount && sagaRepository && saga) {
+              sagaRepository.mount(
+                SagaAbstraction,
+                sagaId,
+                saga,
+                mutatedModel,
+                wrapDispatch(dispatch, action)
+              );
+            } else if (action.type === Unmount && sagaRepository && saga) {
+              sagaRepository.unmount(sagaId);
+            } else if (sagaRepository && saga) {
+              sagaRepository.dispatch(
+                sagaId,
+                mutatedModel,
+                action
+              );
             }
           });
         }
 
-        return reduction;
+        return mutatedModel;
       } else {
-        return model;
+        return reducer(model, action);
```

Comment:
It definitely must have been wrong, because when there's no action provided it should call the underlying reducer to obtain empty model.